### PR TITLE
add import-boss restrictions - staging/src/k8s.io/kube-proxy

### PR DESCRIPTION
**Pull request by saravanan30erd**
**What this PR does / why we need it**:
add import-boss restrictions - staging/src/k8s.io/kube-proxy

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Ref #68201 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by neolit123** [comment]
thanks

/kind cleanup
/sig network
/ok-to-test

**Comment by saravanan30erd** [comment]
/assign @spiffxp 
Could you look on this please?

**Comment by spiffxp** [comment]
@saravanan30erd unfortunately I'm not sure I have enough context to judge what "correct" means for these import-boss PRs, if someone like @dims or @sttts who has a little more experience with this tool and why we use it could demonstrate, I'd be happy to learn how to review this and other PR's

**Comment by saravanan30erd** [comment]
/assign @dims 
cc @nikhita

File: staging/src/k8s.io/kube-proxy/.import-restrictions [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "Rules": [
+    {
+      "SelectorRegexp": "k8s[.]io/apimachinery/",
+      "AllowedPrefixes": [
+        "k8s.io/apimachinery/pkg/runtime",
+        "k8s.io/apimachinery/pkg/runtime/schema",
+        "k8s.io/apimachinery/pkg/apis/config/v1alpha1",
+        "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (sttts) [inline staging/src/k8s.io/kube-proxy/.import-restrictions]: why are we so strict? Why not allow the whole of apimachinery for kube-proxy?

**Comment by spiffxp** [comment]
/remove-lifecycle stale
I still don't really have enough context to review this
/uncc
/cc @dims

**Pull request closed without merging**